addons/crm/models/crm_lead_scoring_frequency.py

```diff
@@ -10,7 +10,7 @@ class LeadScoringFrequency(models.Model):
     value = fields.Char('Value')
     won_count = fields.Float('Won Count', digits=(16, 1))  # Float because we add 0.1 to avoid zero Frequency issue
     lost_count = fields.Float('Lost Count', digits=(16, 1))  # Float because we add 0.1 to avoid zero Frequency issue
-    team_id = fields.Many2one('crm.team', 'Sales Team')
+    team_id = fields.Many2one('crm.team', 'Sales Team', ondelete="cascade")
```

Comment:
Not sure about this one, I would like the opinion of our PLS expert @dbeguin 

I feel like this will make these frequencies statistics "lost" as opposed to considered with "no team", which is a very big difference from a PLS point of view if I remember correctly.

Not sure if setting to NULL would be better as this may not be the way "no team" statistics are handled when computing.
Maybe cascading AND planning a rebuild of the frequency table when a sales.team is deleted would be the way to go? Or maybe the process is too "heavy" and it's not possible?

Anyway, just making sure we are correctly handling this case.

Reply:
"No team" PLS will be computed based on all leads next to an incoming fix in 14.3. Keeping unused statistics is therefore unnecessary, and we will not use those without team_id set. I let @dbeguin confirm :) .

Reply:
Well, yes, keeping ghost team in frequency table is useless.
But, we need to update the frequencies in a way that reflect the lead that had the deleted team to increase the frequency according to the frequency of the deleted team.
Either we do same as won-lost (increasing for each lead), 
OR, we truncate the table and rebuild it (so we won't need to cascade delete the team tin the frequency table anymore, 
OR we take the frequencies of the deleted team and we merge them with the 'team=False' frequencies.
Maybe the last is the better.